subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/GradleBuildExternalPluginsValidationSmokeTest.groovy

```diff
@@ -96,7 +96,8 @@ class GradleBuildExternalPluginsValidationSmokeTest extends AbstractGradleceptio
 
     void validatePlugins() {
         allPlugins.performValidation([
-            "-Dorg.gradle.internal.validate.external.plugins=true"
+            "-Dorg.gradle.internal.validate.external.plugins=true",
+            "--no-parallel" // make sure we have consistent execution ordering as we skip cached tasks
```

Comment:
Is the different execution order/from cache order a problem for the expectations of this test? If it isn't, why do we disable parallel execution?

Reply:
It is (look at the commit message): the test relies on the fact we will get duplicates from cache. If the initial task being executed is not the one we expected, there's a race condition.

Reply:
Can't we change the test, so it doesn't depend on validation warnings to show up in a single project? We do not really care in which project the warning shows up, just that it shows up, right?

Reply:
It's done this way so that we can make sure that new plugins are verified (like Louis faced this morning, he added a plugin and the test failed because we didn't know about it, as expected). If we don't care about the project, then we need a way to collect all answers from all projects, and a way to list all plugins being validated. This is doable but IMO not worth the effort. There's almost no overhead of doing serially here, the build completes roughly in 2 minutes with or without parallel.

Reply:
👍